### PR TITLE
Enable conv+sigmoid fusion (FP32)

### PR DESCRIPTION
**Pull request by liangan1**
Conv + Sigmoid fusion is a key optimization for Squeeze-and-Excitation Networks (https://arxiv.org/pdf/1709.01507.pdf). The representative model is SE-ResNext and its variant. With this fusion, we can reduce the memory footprint for Conv + Sigmoid and therefore improve the performance. As MKL-DNN supports the primitive post op in FP32, we submit the PR to support FP32 first and will add INT8 once MKL-DNN provides the support.

### CODE REVIEW

**Comment by liangan1** [comment]
@sfraczek @wojtuss please help to review this PR. Thanks.

**Comment by liangan1** [comment]
@luotao1   what does  the pr_ci error    "The build was triggered in the branch pull/17103 which does not correspond to any branch monitored by the build VCS roots (the branch is either closed or excluded / not matched by any branch specification). Because of that default branch revisions were set to this build."   means?   The link is [log link](http://ci.paddlepaddle.org/viewLog.html?buildId=90795&tab=buildResultsDiv&buildTypeId=Paddle_PrCi&logTab=)

File: python/paddle/fluid/transpiler/inference_transpiler.py [hunk 1]
```diff
@@ -387,6 +387,37 @@ def _fuse_batch_norm(self, program, place, scope):
...
         program = program.clone()
 
+    def _fuse_conv_sigmoid_mkldnn(self, program):
+        '''
+        Transpile the program by fused sigmoid activation into conv for MKLDNN program.
+        Sigmoid activation following convolution OP can be fused by adding
+        'fuse_sigmoid' attribute to convolution OP.
+        The result of fuse is:
+            - before:
+                - conv -> sigmoid ->any_other_op
```
Comment (sfraczek) [inline python/paddle/fluid/transpiler/inference_transpiler.py:397]: inconsistent formatting. Other comments in this file use no spaces, just arrows.

File: python/paddle/fluid/transpiler/inference_transpiler.py [hunk 2]
```diff
@@ -387,6 +387,37 @@ def _fuse_batch_norm(self, program, place, scope):
...
+            - after:
+                - conv ->any_other_op
```
Comment (sfraczek) [inline python/paddle/fluid/transpiler/inference_transpiler.py:399]: Please remove the space between conv and arrow to keep it consistent with other comments.

**Review by sfraczek — APPROVED** [review]
LGTM.

**Comment by ghost** [comment]
Close without merge. Not in priority and have maintainability concern.

**Pull request closed without merging**